app/services/search/query_builders/chat_channel_membership.rb

```diff
@@ -33,8 +34,14 @@ def initialize(params:)
 
       def build_queries
         @body[:query] = {}
-        @body[:query][:bool] = { filter: filter_conditions }
-        @body[:query][:bool][:must] = query_conditions if query_keys_present?
+        if @params[:user_id] == "all"
+          @body[:query][:bool] = { filter: [{ terms: { status: %w[active joining_request] } }] }
```

Comment:
Rather than having this if statement here I think it would be better to set these params before we send the request to the query builder or [up in the line above that already has a comment](https://github.com/thepracticaldev/dev.to/pull/7385/files#diff-c8d5692235cc1a81cb7ed64d40688e2cR28).

The query builder's sole job is to take a set of params/filters and make an ES query out of it. I want to try as much as possible to not have it being what sets the params/filters which is why I commented on that line above where I set it myself.